ixnetwork_open_traffic_generator/vport.py

```diff
@@ -591,29 +591,40 @@ def _set_ethernet_auto_negotiation(self, vport, layer1, imports):
     def _add_l1config_import(self, vport, proposed_import, imports):
         type = vport['type'].replace('Fcoe', '')
         l1config = vport['l1Config'][type]
+        key_to_remove = []
         for key in proposed_import:
             if key == 'xpath':
                 continue
-            if key in l1config and l1config[key] != proposed_import[key]:
-                imports.append(proposed_import)
-                return
+            if key not in l1config or l1config[key] == proposed_import[key]:
+                key_to_remove.append(key)
+        # add this constrain due to handle some specific use case (1G to 10G)
+        if 'speed' in key_to_remove and 'speedAuto' not in key_to_remove:
+            key_to_remove.remove('speed')
+        for key in key_to_remove:
+            proposed_import.pop(key)
+        if len(proposed_import) > 0:
+            imports.append(proposed_import)
 
     def _set_gigabit_auto_negotiation(self, vport, layer1, imports):
+        advertise = []
+        advertise.append(Vport._SPEED_MAP[layer1.speed])
         proposed_import = {
             'xpath':
             vport['xpath'] + '/l1Config/' + vport['type'].replace('Fcoe', ''),
             'ieeeL1Defaults':
             layer1.ieee_media_defaults,
             'speed':
             Vport._SPEED_MAP[layer1.speed],
-            'enableAutoNegotiation':
+            'autoNegotiate':
```

Comment:
The name of this property is conditional depending on the l1Config -currentType. This method should be expanded to set property names based on the l1Config -currentType.

Reply:
Add validation https://github.com/open-traffic-generator/ixnetwork/pull/210/commits/c4d8c89eb8ace2c13c4c66b6e534859efbe3af41

Please review